File: app/src/main/java/com/nileshdeokar/healthapp/SIngleLongHandler.java
package com.nileshdeokar.healthapp;

/**
 * Created by niel on 27/05/18.
 */

public class SIngleLongHandler {

    private long healthValue = 0L;

    private static final int sizeOfLongInbits = 64;


    public boolean set(int position)
    {
        return setValue(position,true);
    }

    private boolean setValue(int position,boolean value) {

        int offset   = position % sizeOfLongInbits ;
        if(!value) {
            healthValue &= ~ (1L << offset);
        }
        else
        {
            healthValue |= 1L << offset;

        }
        System.out.println("set : "+position + " " + value);
        return true;
    }

    public boolean get(int position) {

        int offset   = position % sizeOfLongInbits;
        long bit = (healthValue >> offset) & 1L;
        System.out.println("get : "+position + " " + (bit!=0));
        return bit!=0;
    }

    public void clear(int position)
    {
        setValue(position,false);
    }

    public long toLongs()
    {
        return this.healthValue;
    }

    public void setLongvalue(long longvalue){
        healthValue = longvalue;
    }

    public void toggleDisease(int diseaseToSet, boolean value) {
        if(value) set(diseaseToSet); else clear(diseaseToSet);
    }
}

File: app/src/main/java/com/nileshdeokar/healthapp/utils/SingleLongHandler.kt
package com.nileshdeokar.healthapp.utils

/**
 * Created by niel on 27/05/18.
 */

class SingleLongHandler {

    private var healthValue = 0L


    fun set(position: Int): Boolean {
        return setValue(position, true)
    }

    private fun setValue(position: Int, value: Boolean): Boolean {

        val offset = position % sizeOfLongInbits
        if (!value) {
            healthValue = healthValue and (1L shl offset).inv()
        } else {
            healthValue = healthValue or (1L shl offset)

        }
        println("set : $position $value")
        return true
    }

    operator fun get(position: Int): Boolean {

        val offset = position % sizeOfLongInbits
        val bit = healthValue shr offset and 1L
        println("get : " + position + " " + (bit != 0L))
        return bit != 0L
    }

    fun clear(position: Int) {
        setValue(position, false)
    }

    fun toLongs(): Long {
        return this.healthValue
    }

    fun setLongvalue(longvalue: Long) {
        healthValue = longvalue
    }

    fun toggleDisease(diseaseToSet: Int, value: Boolean) {
        if (value) set(diseaseToSet) else clear(diseaseToSet)
    }

    companion object {

        private const val sizeOfLongInbits = 64
    }
}
